woocommerce/woocommerce-ios
# [My Store] Add support for conditionally storing site summary stats

Part of #8173
⚠️ Depends on #8420 ⚠️ 


## Description


This PR adds support in the Yosemite layer for conditionally storing `SiteSummaryStats` fetched from remote:

* Adds support for retrieving `SiteSummaryStats` from storage.
* Adds a `saveInStorage` parameter to `StatsActionV4.retrieveSiteSummaryStats` to determine if stats should be stored.
* Updates `StatsStoreV4` to support conditionally upserting stats fetched from remote.

Note: For now we only store `SiteSummaryStats` that are fetched for a single period. The My Store dashboard only needs summary stats for a single period (day, week, month, year) and stats fetched for multiple periods don't need to be stored (used only in the Analytics Hub, for quarter periods).

## Testing instructions


1. Build and run the app.
2. Tap "See more" to open Analytics.
3. Confirm the Sessions card loads as expected for different time ranges.

---
